### DISCUSSION TITLE
Custom CV generator - plain iterator

### DISCUSSION THREAD

**Discussion by lhaffonso** [pycaret/pycaret#1488] [Ideas]
Hello,

Is there any plan to introduce a plain iterator of indices instead of a sklearn cv generator object? I've managed to do it in Catboost.
My use case is multiple time series at once so TimeSeriesSplit doesn't suit my needs since there are repetition of rows in same dates.

Thank you

**Comment by Yard1** [comment]
You can pass any sklearn-compatible CV object/generator as the `cv` argument.

**Reply by lhaffonso** [reply]
@Yard1 Thanks for your reply. Yes I know, for my case is not practical to use a sklearn cv generator object.
Catboost has this option - I don't have to have a split method for example, check it out:
https://catboost.ai/docs/concepts/python-reference_cv.html
